ibis/util.py

```diff
@@ -355,3 +356,39 @@ def flatten_iterable(iterable):
             yield from flatten_iterable(item)
         else:
             yield item
+
+
+def warn_deprecated(name, ver='', instead='', ignoreframes=1):
+    """Warn about deprecated usage, including stacktrace, and
+    what to do instead."""
+    warnlines = []
+    msg = f'"{name}" is deprecated'
+    if ver:
+        msg += f' since v{ver}'
+
+    if instead:
+        msg += f'; use {instead}.'
+
+    warnlines.append(msg)
+
+    import traceback
+
+    warnlines.append('Deprecated call traceback (most recent last):')
+    for f in traceback.extract_stack(limit=6)[:-ignoreframes]:
```

Comment:
This hardcoded 6 feels a bit arbitrary, any particular reason why 6?

Reply:
Probably better would be `ignoredframes+N`, where 1 <= N <= 5.  A full stacktrace is unwieldy, but it can be helpful to get more context than just 1 frame would provide. I would choose N=3 as a simple heuristic.  N=1 is also acceptable though if you want to keep logging noise to a minimum.

Reply:
I'm thinking of removing the stacktrace entirely and just passing `ignoreframes+1` to `warnings.warn`.